File: solutions/javascript/nucleotide-count/1/nucleotide-count.js
function countNucleotides(strand) {
  const nucleotideCounts = { A: 0, C: 0, G: 0, T: 0 };

  for (const nucleotide of strand) {
    if (!(nucleotide in nucleotideCounts)) {
      throw new Error('Invalid nucleotide in strand');
    }
    nucleotideCounts[nucleotide]++;
  }

  return `${nucleotideCounts.A} ${nucleotideCounts.C} ${nucleotideCounts.G} ${nucleotideCounts.T}`;
}

export { countNucleotides };

File: solutions/julia/nucleotide-count/1/nucleotide-count.jl
"""
    count_nucleotides(strand)

The count of each nucleotide within `strand` as a dictionary.

Invalid strands raise a `DomainError`.

"""
function count_nucleotides(strand)
    result = Dict('A' => 0, 'C' => 0, 'G' => 0, 'T' => 0)
    [haskey(result, key) ? result[key] += 1 : throw(DomainError(key)) for key in strand]
    return result
end
